### PR TITLE
Adds detailed description about io trace

### PR DESCRIPTION
**Pull request by mariuszbarczak**
Signed-off-by: Mariusz Barczak <mariusz.barczak@intel.com>

### CODE REVIEW

File: doc/IOTRACER.md [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+information is captured (IO operation type, address, size), supplemented with
+extended classification. Extended classification contains information about I/O
+type (direct / filesystem metadata / file) and target file attributes(e.g. file
+size). Afterwards traces can later be converted to JSON or CSV format and
+consumed by any further analytics. iotrace does it and extends outputs with
+post-processing fields like IO latency, IO queue depth, and more.
+
+iotrace is based on [Open CAS Telemetry Framework (OCTF)](https://github.com/Open-CAS/open-cas-telemetry-framework). Collected traces are stored in format defined by OCTF using google protocol
+buffer description. You can find this definition [here](https://github.com/Open-CAS/open-cas-telemetry-framework/blob/master/source/octf/proto/trace.proto). Such trace format allows to analyze
+it by any kind of tool like Java, C++, Python, etc.,
```
Comment (tomaszrybicki) [inline doc/IOTRACER.md]: I suggest "any kind of tool written in Java, C++, Python, or other languages( supported by Protocol Buffer)"

File: doc/IOTRACER.md [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+
+The goodness of OCTF can be reused by other storage application for instance
```
Comment (tomaszrybicki) [inline doc/IOTRACER.md]: applications

File: doc/IOTRACER.md [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+SKDP, QEMU. They can produce IO traces in the same OCTF traces format, what can
```
Comment (tomaszrybicki) [inline doc/IOTRACER.md]: SPDK?

Comment (tomaszrybicki) [inline doc/IOTRACER.md]: which can

File: doc/IOTRACER.md [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+be used by the same analytics tools. But that's next steps.
```
Comment (tomaszrybicki) [inline doc/IOTRACER.md]: But these are

File: doc/IOTRACER.md [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+
+## Getting traces
+
+The iotrace [README](https://github.com/Open-CAS/standalone-linux-io-tracer/blob/master/README.md)
+contains basic information how to install iotrace on your machine. Please follow
+this instruction. Let's get a trace. Any time you can call the tool help:
```
Comment (tomaszrybicki) [inline doc/IOTRACER.md]: tool's

File: doc/IOTRACER.md [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+
+Options that are valid with {-S | --start-trace}
+     -b    --buffer <1-1024>                     Size of the internal trace buffer (in MiB) (default: 100)
+     -d    --devices <VALUE>[,VALUE]             Paths of devices to be traced
+     -l    --label <VALUE>                       User defined label
+     -s    --size <1-100000000>                  Max size of trace file (in MiB) (default: 1000)
+     -t    --time <1-4294967295>                 Max trace duration time (in seconds) (default: 4294967295)
+~~~
+
+Let's say, your workload/application is running on top of two two block devices.
```
Comment (tomaszrybicki) [inline doc/IOTRACER.md]: two two

File: doc/IOTRACER.md [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+We would like to trace them for 1 hour, or until trace file is 1GiB. Also you
+can label the trace with you name. We are going to call:
```
Comment (tomaszrybicki) [inline doc/IOTRACER.md]: your name

File: doc/IOTRACER.md [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+
+~~~{.sh}
+sudo iotrace --start-trace --devices /dev/nvme0n1,/dev/nvme1n1 --time 3600 --size 1024 --label "My cool first trace"
+~~~
+
+But any time you can interrupt trace collection by pressing Ctrl+C or sending
```
Comment (tomaszrybicki) [inline doc/IOTRACER.md]: the trace collection

File: doc/IOTRACER.md [hunk 9]
```diff
@@ -0,0 +1,304 @@
...
+Let's say, your workload/application is running on top of two two block devices.
+We would like to trace them for 1 hour, or until trace file is 1GiB. Also you
+can label the trace with you name. We are going to call:
+
+~~~{.sh}
+sudo iotrace --start-trace --devices /dev/nvme0n1,/dev/nvme1n1 --time 3600 --size 1024 --label "My cool first trace"
+~~~
+
+But any time you can interrupt trace collection by pressing Ctrl+C or sending 
+termination signal to the iotrace process. When stop tracing you should get
```
Comment (tomaszrybicki) [inline doc/IOTRACER.md]: After tracing has stopped you should

File: doc/IOTRACER.md [hunk 10]
```diff
@@ -0,0 +1,304 @@
...
+Remember your trace path **kernel/2019-08-13_12:35:22**. We will use it for further
+processing.
+
+## Parsing traces
+
+The iotrace parser converts IO traces from binary format to CSV or JSON format.
+To parse you trace we are going to invoke _--parse-trace_ command.
+
+~~~{.sh}
+otrce --parse-trace --help
```
Comment (tomaszrybicki) [inline doc/IOTRACER.md]: iotrace

File: doc/IOTRACER.md [hunk 11]
```diff
@@ -0,0 +1,304 @@
...
+...
+{"header":{"sid":"18834","timestamp":"134120569337"},"io":{"lba":"1465145088","len":88,"ioClass":14,"operation":"Write","flush":false,"fua":false,"error":false,"latency":"32495","qd":"2"},"device":{"id":"271581185","name":"nvme0n1"},"file":{"id":"1610612801","offset":"112","size":"192"}}
+{"header":{"sid":"18838","timestamp":"134120638297"},"io":{"lba":"904","len":120,"ioClass":14,"operation":"Write","flush":false,"fua":false,"error":false,"latency":"40076","qd":"1"},"device":{"id":"271581185","name":"nvme0n1"},"file":{"id":"73","offset":"0","size":"203"}}
+{"header":{"sid":"18840","timestamp":"134120664253"},"io":{"lba":"1024","len":88,"ioClass":14,"operation":"Write","flush":false,"fua":false,"error":false,"latency":"30424","qd":"2"},"device":{"id":"271581185","name":"nvme0n1"},"file":{"id":"73","offset":"120","size":"203"}}
+{"header":{"sid":"18844","timestamp":"134120707517"},"io":{"lba":"488382416","len":48,"ioClass":14,"operation":"Write","flush":false,"fua":false,"error":false,"latency":"20330","qd":"1"},"device":{"id":"271581185","name":"nvme0n1"},"file":{"id":"536870985","offset":"0","size":"193"}}
+...
+~~~
+
+## What we collect, What we process?
```
Comment (tomaszrybicki) [inline doc/IOTRACER.md]: What do we collect, what do we process

File: doc/IOTRACER.md [hunk 12]
```diff
@@ -0,0 +1,304 @@
...
+
+The bellow table contains telemetry content which is traces by iotrace. Also we
```
Comment (tomaszrybicki) [inline doc/IOTRACER.md]: below

Comment (tomaszrybicki) [inline doc/IOTRACER.md]: traced

File: doc/IOTRACER.md [hunk 13]
```diff
@@ -0,0 +1,304 @@
...
+included future plans for tracing.
```
Comment (tomaszrybicki) [inline doc/IOTRACER.md]: We also included our future plans

File: doc/IOTRACER.md [hunk 14]
```diff
@@ -0,0 +1,304 @@
...
+.tg th{font-family:Arial, sans-serif;font-size:14px;font-weight:normal;padding:10px 5px;border-style:solid;border-width:1px;overflow:hidden;word-break:normal;border-color:#9ABAD9;color:#fff;background-color:#409cff;}
+.tg .tg-j3py{background-color:#D2E4FC;border-color:#000000;text-align:left;vertical-align:top}
+.tg .tg-mcqj{font-weight:bold;border-color:#000000;text-align:left;vertical-align:top}
+.tg .tg-73oq{border-color:#000000;text-align:left;vertical-align:top}
+</style>
+<table class="tg">
+  <tr>
+    <th class="tg-mcqj">Category</th>
+    <th class="tg-mcqj">Telemetry Content</th>
+    <th class="tg-mcqj">Post processing information</th>
```
Comment (tomaszrybicki) [inline doc/IOTRACER.md]: Inconsistent casing of words (in whole table)

File: doc/IOTRACER.md [hunk 15]
```diff
@@ -0,0 +1,304 @@
...
+    <th class="tg-mcqj">Status</th>
+  </tr>
+  <tr>
+    <td class="tg-j3py">Basic IO metdata</td>
```
Comment (tomaszrybicki) [inline doc/IOTRACER.md]: Metadata

File: doc/IOTRACER.md [hunk 16]
```diff
@@ -0,0 +1,304 @@
...
+The bellow table contains telemetry content which is traces by iotrace. Also we
+included future plans for tracing.
+
+<style type="text/css">
+.tg  {border-collapse:collapse;border-spacing:0;border-color:#9ABAD9;}
+.tg td{font-family:Arial, sans-serif;font-size:14px;padding:10px 5px;border-style:solid;border-width:1px;overflow:hidden;word-break:normal;border-color:#9ABAD9;color:#444;background-color:#EBF5FF;}
+.tg th{font-family:Arial, sans-serif;font-size:14px;font-weight:normal;padding:10px 5px;border-style:solid;border-width:1px;overflow:hidden;word-break:normal;border-color:#9ABAD9;color:#fff;background-color:#409cff;}
+.tg .tg-j3py{background-color:#D2E4FC;border-color:#000000;text-align:left;vertical-align:top}
+.tg .tg-mcqj{font-weight:bold;border-color:#000000;text-align:left;vertical-align:top}
+.tg .tg-73oq{border-color:#000000;text-align:left;vertical-align:top}
```
Comment (tomaszrybicki) [inline doc/IOTRACER.md]: Please check if this shows up properly - it doesn't work in github diff (when switched to markdown viewer)

**Comment by mariuszbarczak** [comment]
This should be pull requested on iotrace

**Pull request closed without merging**